homeassistant/components/kostal_plenticore/sensor.py

```diff
@@ -0,0 +1,192 @@
+    settings_data_update_coordinator = SettingDataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        "Settings Data",
+        timedelta(seconds=300),
+        plenticore,
+    )
+    for module_id, data_id, name, sensor_data, fmt in SENSOR_SETTINGS_DATA:
+        if module_id not in available_settings_data or data_id not in (
+            setting.id for setting in available_settings_data[module_id]
+        ):
+            _LOGGER.debug(
+                "Skipping non existing setting data %s/%s", module_id, data_id
+            )
+            continue
+
+        entities.append(
+            PlenticoreDataSensor(
+                settings_data_update_coordinator,
+                entry.entry_id,
+                entry.title,
+                module_id,
+                data_id,
+                name,
+                sensor_data,
+                PlenticoreDataFormatter.get_method(fmt),
+                plenticore.device_info,
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class PlenticoreDataSensor(CoordinatorEntity):
+    """Representation of a Plenticore data Sensor."""
+
+    def __init__(
+        self,
+        coordinator,
+        entry_id: str,
+        platform_name: str,
+        module_id: str,
+        data_id: str,
+        sensor_name: str,
+        sensor_data: Dict[str, Any],
+        formatter: Callable[[str], Any],
+        device_info: Dict[str, Any],
+    ):
+        """Create a new Sensor Entity for Plenticore process data."""
+        super().__init__(coordinator)
+        self.entry_id = entry_id
+        self.platform_name = platform_name
+        self.module_id = module_id
+        self.data_id = data_id
+
+        self._sensor_name = sensor_name
+        self._sensor_data = sensor_data
+        self._formatter = formatter
+
+        self._device_info = device_info
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return (
+            super().available
+            and self.coordinator.data is not None
+            and self.module_id in self.coordinator.data
+            and self.data_id in self.coordinator.data[self.module_id]
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Register this entity on the Update Coordinator."""
+        await super().async_added_to_hass()
+        self.coordinator.start_fetch_data(self.module_id, self.data_id)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Unregister this entity from the Update Coordinator."""
+        self.coordinator.stop_fetch_data(self.module_id, self.data_id)
+        await super().async_will_remove_from_hass()
+
+    @property
+    def device_info(self) -> Dict[str, Any]:
+        """Return the device info."""
+        return self._device_info
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique id of this Sensor Entity."""
+        return f"{self.entry_id}_{self._sensor_name}"
```

Comment:
I suggest we use `self.data_id` as suffix. It's a bit safer. If we ever want to change the name of a sensor we don't need to change the unique id.

Reply:
good point!